<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Sequence Robust Association Test (SRAT) • SRAT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Sequence Robust Association Test (SRAT)">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SRAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SRAT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/celehs/SRAT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Sequence Robust Association Test (SRAT)</h1>
            <h3 data-toc-skip class="subtitle">A Rank-Based Method for
Testing Genetic Associations with Familial Data</h3>
            
      
      <small class="dont-index">Source: <a href="https://github.com/celehs/SRAT/blob/HEAD/vignettes/SRAT.Rmd" class="external-link"><code>vignettes/SRAT.Rmd</code></a></small>
      <div class="hidden name"><code>SRAT.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The Sequence Robust Association Test (SRAT) is a fully rank-based,
flexible approach to test for association between a set of genetic
variants and an outcome, while accounting for within-family correlation
and adjusting for covariates. SRAT was developed specifically for
genetic association studies that involve family data, such as
genome-wide association studies (GWAS) and next-generation sequencing
studies (NGSS).</p>
<div class="section level3">
<h3 id="key-features-and-advantages">Key Features and Advantages<a class="anchor" aria-label="anchor" href="#key-features-and-advantages"></a>
</h3>
<p>SRAT offers several advantages over traditional methods:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Robustness</strong>: As a fully rank-based method, SRAT
is robust to outliers and non-normality in the outcome
distribution.</p></li>
<li><p><strong>Flexibility in correlation structure</strong>: SRAT
allows for unknown correlation structures within families.</p></li>
<li><p><strong>Improved type I error control</strong>: SRAT provides
better protection against type I error rate inflation compared to
existing methods.</p></li>
<li><p><strong>Enhanced power</strong>: For settings with skewed outcome
distributions, SRAT can be more powerful than parametric
approaches.</p></li>
<li><p><strong>Special case</strong>: SRAT includes the well-known
Wilcoxon rank sum test as a special case.</p></li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="basic-usage">Basic Usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>First, load the required packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://celehs.github.io/SRAT/" class="external-link">SRAT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SKAT</span><span class="op">)</span>  <span class="co"># For comparison and example data</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>For demonstration, we’ll use the example data from the SKAT package,
which contains: - <code>Z</code>: A genotype matrix containing genetic
variants - <code>X</code>: A covariate matrix (e.g., age, gender) -
<code>y.c</code>: A continuous phenotype variable</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SKAT.example</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/attach.html" class="external-link">attach</a></span><span class="op">(</span><span class="va">SKAT.example</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="comparing-skat-and-srat">Comparing SKAT and SRAT<a class="anchor" aria-label="anchor" href="#comparing-skat-and-srat"></a>
</h3>
<p>Let’s first run the analysis using SKAT for comparison:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run SKAT analysis</span></span>
<span><span class="va">obj.skat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SKAT/man/SKAT_Null_Model.html" class="external-link">SKAT_Null_Model</a></span><span class="op">(</span><span class="va">y.c</span> <span class="op">~</span> <span class="va">X</span>, out_type <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">skat_pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SKAT/man/SKAT.html" class="external-link">SKAT</a></span><span class="op">(</span><span class="va">Z</span>, <span class="va">obj.skat</span>, weights.beta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"SKAT p-value:"</span>, <span class="va">skat_pval</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now, let’s run the same analysis using SRAT:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run SRAT analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>  <span class="co"># For reproducibility</span></span>
<span><span class="va">obj.srat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/srat.null.html">srat.null</a></span><span class="op">(</span><span class="va">y.c</span>, <span class="va">X</span><span class="op">)</span>  <span class="co"># Fit null model</span></span>
<span><span class="va">srat_pval</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/srat.test.html">srat.test</a></span><span class="op">(</span><span class="va">Z</span>, <span class="va">obj.srat</span><span class="op">)</span>  <span class="co"># Test for association</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"SRAT p-value:"</span>, <span class="va">srat_pval</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="srat-workflow-explained">SRAT Workflow Explained<a class="anchor" aria-label="anchor" href="#srat-workflow-explained"></a>
</h2>
<p>SRAT operates in a two-step process:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Null model fitting</strong> (<code><a href="../reference/srat.null.html">srat.null()</a></code>):
This function fits a null model that accounts for the covariates while
ignoring the genetic variants. It returns an object containing elements
needed for the subsequent testing.</p></li>
<li><p><strong>Association testing</strong> (<code><a href="../reference/srat.test.html">srat.test()</a></code>):
This function tests for the association between the genetic variants and
the outcome, conditional on the null model.</p></li>
</ol>
<div class="section level3">
<h3 id="parameters-and-options">Parameters and Options<a class="anchor" aria-label="anchor" href="#parameters-and-options"></a>
</h3>
<div class="section level4">
<h4 id="srat-null">srat.null()<a class="anchor" aria-label="anchor" href="#srat-null"></a>
</h4>
<ul>
<li>
<code>y</code>: Outcome vector (e.g., phenotype)</li>
<li>
<code>X</code>: Covariates for adjustment (e.g., age, gender)</li>
<li>
<code>B</code>: Number of perturbations (default: 1000)</li>
</ul>
</div>
<div class="section level4">
<h4 id="srat-test">srat.test()<a class="anchor" aria-label="anchor" href="#srat-test"></a>
</h4>
<ul>
<li>
<code>Z</code>: Covariates for testing (e.g., genotype)</li>
<li>
<code>obj</code>: Object from null model</li>
<li>
<code>w.sqrt</code>: Square root of weight vector (default: all
1’s)</li>
</ul>
</div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-example-using-weights">Advanced Example: Using Weights<a class="anchor" aria-label="anchor" href="#advanced-example-using-weights"></a>
</h2>
<p>SRAT allows for weighting of genetic variants, which can improve
power when certain variants are more likely to be causal:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create weights (example: upweight rare variants)</span></span>
<span><span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">Z</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">Z</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span></span>
<span><span class="va">rare_variants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">maf</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">weights</span><span class="op">[</span><span class="va">rare_variants</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span></span>
<span><span class="co"># Apply weights in SRAT</span></span>
<span><span class="va">obj.srat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/srat.null.html">srat.null</a></span><span class="op">(</span><span class="va">y.c</span>, <span class="va">X</span><span class="op">)</span></span>
<span><span class="va">weighted_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/srat.test.html">srat.test</a></span><span class="op">(</span><span class="va">Z</span>, <span class="va">obj.srat</span>, w.sqrt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"SRAT p-value with weights:"</span>, <span class="va">weighted_result</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="understanding-the-output">Understanding the Output<a class="anchor" aria-label="anchor" href="#understanding-the-output"></a>
</h2>
<p>The p-value returned by SRAT indicates the statistical significance
of the association between the set of genetic variants and the
phenotype. A smaller p-value suggests stronger evidence against the null
hypothesis of no association.</p>
</div>
<div class="section level2">
<h2 id="comparison-with-other-methods">Comparison with Other Methods<a class="anchor" aria-label="anchor" href="#comparison-with-other-methods"></a>
</h2>
<p>SRAT has been shown to be particularly advantageous when: - The
outcome distribution is skewed or contains outliers - Family correlation
structures are complex - The sample size is relatively small</p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>SRAT provides a robust alternative to existing methods for testing
genetic associations in family-based studies. Its rank-based nature and
ability to handle family correlation make it suitable for a wide range
of genetic studies.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Dai W, Yang M, Wang C, Cai T. Sequence robust association test for
familial data. Biometrics. 2017 Sep;73(3):876-884. doi:
10.1111/biom.12643.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ming Yang, Wei Dai, CELEHS Lab, PARSE Health.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
